pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -903,6 +918,23 @@ func getFailingSyncSets(syncStatuses []hiveintv1alpha1.SyncStatus) []string {
 	return failures
 }
 
+func (r *ReconcileClusterSync) setFirstSuccessTime(syncStatuses []hiveintv1alpha1.SyncStatus, cd *hivev1.ClusterDeployment, clusterSync *hiveintv1alpha1.ClusterSync, logger log.FieldLogger) {
+	lastSuccessTime := &metav1.Time{}
+	for _, status := range syncStatuses {
+		if status.FirstSuccessTime == nil {
+			return
+		}
+		if status.FirstSuccessTime.Time.After(lastSuccessTime.Time) {
+			lastSuccessTime = status.FirstSuccessTime
+		}
+	}
+	clusterSync.Status.FirstSuccessTime = lastSuccessTime
+	allSyncSetsAppliedDuration := lastSuccessTime.Time.Sub(cd.Status.InstalledTimestamp.Time)
```

Comment:
There needs to be a check that `cd.Status.InstalledTimestamp` is not nil. The field gets set in the clusterDeployment controller, but we cannot guaranteed that that has been done at this point.

Reply:
Added a check at the top of the function.